src/lib.rs

```diff
@@ -0,0 +1,230 @@
+            }
+        }
+        None => Err(anyhow!("HHA is not installed")),
+    }
+}
+
+#[instrument(err, fields(path = %path.as_ref().display()))]
+pub fn load_mem_proof_file(path: impl AsRef<Path>) -> Result<HashMap<String, MembraneProof>> {
+    use std::fs::File;
+
+    let file = File::open(path).context("failed to open file")?;
+    let proof: MembraneProofFile =
+        serde_yaml::from_reader(&file).context("failed to deserialize YAML as MembraneProof")?;
+    debug!(?proof);
+    proof
+        .payload
+        .into_iter()
+        .map(|p| {
+            base64::decode(p.proof.clone())
+                .map(|proof| (p.cell_nick, MembraneProof::from(UnsafeBytes::from(proof))))
+                .map_err(|e| anyhow!("failed to decode proof: {:?}", e))
+        })
+        .collect()
+}
+
+#[instrument(err, fields(path = %path.as_ref().display()))]
+pub fn load_happ_file(path: impl AsRef<Path>) -> Result<HappsFile> {
+    use std::fs::File;
+
+    let file = File::open(path).context("failed to open file")?;
+    let happ_file =
+        serde_yaml::from_reader(&file).context("failed to deserialize YAML as HappsFile")?;
+    debug!(?happ_file);
+    Ok(happ_file)
+}
+
+#[instrument(err, skip(url))]
+pub(crate) async fn download_file(url: &Url) -> Result<PathBuf> {
+    use isahc::config::RedirectPolicy;
+    use isahc::prelude::*;
+
+    let path = if url.scheme() == "file" {
+        let p = PathBuf::from(url.path());
+        debug!("Using: {:?}", p);
+        p
+    } else {
+        debug!("downloading");
+        let mut url = Url::clone(&url);
+        url.set_scheme("https")
+            .map_err(|_| anyhow!("failed to set scheme to https"))?;
+        let client = HttpClient::builder()
+            .redirect_policy(RedirectPolicy::Follow)
+            .build()
+            .context("failed to initiate download request")?;
+        let mut response = client
+            .get(url.as_str())
+            .context("failed to send GET request")?;
+        if !response.status().is_success() {
+            return Err(anyhow!(
+                "response status code {} indicated failure",
+                response.status().as_str()
+            ));
+        }
+        let dir = TempDir::new().context("failed to create tempdir")?;
+        let url_path = PathBuf::from(url.path());
+        let basename = url_path
+            .file_name()
+            .context("failed to get basename from url")?;
+        let path = dir.into_path().join(basename);
+        let mut file = fs::File::create(&path).context("failed to create target file")?;
+        response
+            .copy_to(&mut file)
+            .context("failed to write response to file")?;
+        debug!("download successful");
+        path
+    };
+    Ok(path)
+}
+
+// Returns true if app should be kept active in holochain
+fn _keep_app_active(installed_app_id: &str, happs_to_keep: Vec<String>) -> bool {
```

Comment:
Seems like this isn't being used. Do we still need it?

Reply:
I kinda kept it in to remind us that we should probably also build in a feature to deactivate happs disabled happs based on hha. But I guess I can just write a note somewhere about it :sweat_smile: